packages/lib/src/test_utils/hooks.ts

```diff
@@ -64,4 +64,30 @@ const useEffectDebugger = (
   useEffect(effectHook, dependencies);
 };
 
-export { useEffectDebugger };
+/**
+ * `renderComponentWithHook` is a utility function that renders a component with a given hook for use in testing
+ * @param useHook - The hook to use
+ * @param Comp - The component to render
+ * @returns - The component and the hook
+ */
+const renderComponentWithHook = <
+  T extends () => any,
+  U extends JSX.Element | null,
+>(
```

Comment:
Does this belong in the `hooks` folder? The other functions in this file are true hooks, but this is a render function to help unit testing hooks. Maybe it belongs in `index`?

Reply:
Good call out!